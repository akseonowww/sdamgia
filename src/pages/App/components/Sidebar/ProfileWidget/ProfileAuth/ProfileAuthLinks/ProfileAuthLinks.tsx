import React, { FC, memo } from 'react'

import LinkWrap from 'components/Link/_wrap'

interface ILinksProps {
  blockClassName: string
}

const ProfileAuthLinks: FC<ILinksProps> = ({ blockClassName }) => {
  const links = [
    {
      href: '/register',
      text: 'Зарегистрироваться',
    },
    {
      href: '/pass_change',
      text: 'Восстановление пароля',
    },
    {
      href: '/vkauth',
      text: 'Вход через Вконтакте',
    },
  ]

  return (
    <div className={`${blockClassName}-Links`}>
      {links.map(({ href, text }, i) => (
        <LinkWrap
          className={`${blockClassName}-Link`}
          href={href}
          key={`${blockClassName}-Link_${i}`}
        >
          {text}
        </LinkWrap>
      ))}
    </div>
  )
}

export default memo(ProfileAuthLinks)
