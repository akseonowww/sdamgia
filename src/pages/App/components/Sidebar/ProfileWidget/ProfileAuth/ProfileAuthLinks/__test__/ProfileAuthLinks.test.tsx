import React from 'react'
import { unmountComponentAtNode, render } from 'react-dom'
import { act } from 'react-dom/test-utils'
import pretty from 'pretty'
import { cn } from '@bem-react/classname'

import ProfileAuthLinks from '../ProfileAuthLinks'

describe('ProfileAuthLinks', () => {
  const profileAuth = cn('ProfileAuth')
  let container: HTMLElement | null = null

  beforeEach(() => {
    container = document.createElement('div')
    document.body.appendChild(container)
  })

  afterEach(() => {
    if (container) {
      unmountComponentAtNode(container)
      container.remove()
      container = null
    }
  })

  it('renders with required and optional props', () => {
    act(() => {
      render(<ProfileAuthLinks blockClassName={profileAuth()} />, container)
    })
    expect(container && container.textContent).toBe(
      'Восстановление пароляВход через ВконтактеРегистрация'
    )
  })

  it('matches snapshot', () => {
    act(() => {
      render(<ProfileAuthLinks blockClassName={profileAuth()} />, container)
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()
  })
})
