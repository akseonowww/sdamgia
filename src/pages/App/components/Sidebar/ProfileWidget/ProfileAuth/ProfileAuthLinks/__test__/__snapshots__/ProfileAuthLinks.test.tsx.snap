// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileAuthLinks matches snapshot 1`] = `"<div class=\\"ProfileAuth-Links\\"><a class=\\"Link Link_wrap ProfileAuth-Link\\" href=\\"/register\\" rel=\\"noopener noreferrer\\"><u class=\\"Link-U Link_wrap-U\\">Зарегистрироваться</u></a><a class=\\"Link Link_wrap ProfileAuth-Link\\" href=\\"/pass_change\\" rel=\\"noopener noreferrer\\"><u class=\\"Link-U Link_wrap-U\\">Восстановление пароля</u></a><a class=\\"Link Link_wrap ProfileAuth-Link\\" href=\\"/vkauth\\" rel=\\"noopener noreferrer\\"><u class=\\"Link-U Link_wrap-U\\">Вход через Вконтакте</u></a></div>"`;
